### PR TITLE
Add `ArrayrHelper::renameKey()`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/ArrayHelper.php [hunk 1]
```diff
@@ -1330,6 +1330,29 @@ public static function getObjectVars(object $object): array
...
+     * @return array The result array.
+     */
+    public static function replaceKey(array $array, int|string $from, int|string $to): array
+    {
+        if (!isset($array[$from])) {
+            return $array;
+        }
+
+        $result = [];
+        foreach ($array as $key => $value) {
```
Comment (samdark) [inline src/ArrayHelper.php:1349]: I don't think `foreach` is needed here. Also, it doesn't remove old key.

```php
if (array_key_exists($from, $result)) {
   $result[$to] = $value;
   unset($result['from'])
}
```

File: src/ArrayHelper.php [hunk 2]
```diff
@@ -1330,6 +1330,29 @@ public static function getObjectVars(object $object): array
...
+    /**
+     * Replace key in array.
+     *
+     * @param array $array Source array.
+     * @param int|string $from Key to replace.
+     * @param int|string $to Key to replace with.
+     *
+     * @return array The result array.
+     */
+    public static function replaceKey(array $array, int|string $from, int|string $to): array
```
Comment (samdark) [inline src/ArrayHelper.php:1342]: How about nested array keys provided with `.` syntax?

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ArrayHelper.php (hunk 1 above)
Comment (vjik) [reply src/ArrayHelper.php:1349]: This approach change order of items.

File: src/ArrayHelper.php (hunk 2 above)
Comment (vjik) [reply src/ArrayHelper.php:1342]: I suggest we leave this for future improvements.

File: src/ArrayHelper.php (hunk 1 above)
Comment (samdark) [reply src/ArrayHelper.php:1349]: Right. The code makes sense then.

File: src/ArrayHelper.php (hunk 2 above)
Comment (samdark) [reply src/ArrayHelper.php:1342]: OK/

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Should we name it `renameKey()`?

**Comment by vjik** [comment]
> Should we name it `renameKey()`?

I don't know. For me both names are OK.

**Comment by samdark** [comment]
I'd make it `renameKey()`.

File: src/ArrayHelper.php [hunk 3]
```diff
@@ -1330,6 +1330,29 @@ public static function getObjectVars(object $object): array
...
+        if (!isset($array[$from])) {
+            return $array;
+        }
+
+        $result = [];
+        foreach ($array as $key => $value) {
+            $result[$key === $from ? $to : $key] = $value;
+        }
+
+        return $result;
```
Comment (Tigrov) [inline src/ArrayHelper.php:1353]:
```suggestion
        $keys = array_keys($array);
        $index = array_search($from, $keys);
        $keys[$index] = $to;
        
        return array_combine($keys, $array);
```

**Pull request merged**